PlayerController.cs

```diff
@@ -0,0 +1,321 @@
+﻿using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.Timeline;
+
+public class PlayerController : MonoBehaviour
+{
+    public PlayerStats stats;
+    private bool gameOver = false;
+    private Animator blackScreenAnimator;
+
+    private static bool flip = true;
+
+    private SaveManager saveManager;
+
+    private bool isInDangerPlatform = false;
+    private bool isInHealPlatform = false;
+    public AudioSource oofSound;
+    public AudioSource healSound;
+
+    public int playerMaxHealth = 100;
+    public int playerMaxShield = 100;
+    public HealthBar hpBar;
+    public int currentHealth;
+    public int currentShield;
+
+    public float speed;
+    public float jumpForce;
+    private float moveInput;
+
+    private static Rigidbody2D rb2d;
+    private bool facingRight;
+
+    private bool isGrounded;
+    public Transform groundCheck;
+    public float checkRadius;
+    public LayerMask whatIsGround;
+
+    private int extraJumps;
+    public int extraJumpsValue;
```

Comment:
Ask yourself why public variables/fields need to be so. They are public if the are going to be exposed to the outside world.

Again, I group by access modifier to make it more readable. I separete it by emtpy line. Like so.

```suggestion
    public PlayerStats stats;
    public AudioSource oofSound;
    public AudioSource healSound;
    public int playerMaxHealth = 100;
    public int playerMaxShield = 100;
    public int currentHealth;
    public int currentShield;
    public int extraJumpsValue;
    public HealthBar hpBar;
    public float speed;
    public float jumpForce;
    public Transform groundCheck;
    public float checkRadius;
    public LayerMask whatIsGround;
    
    private static bool flip = true;
    private static Rigidbody2D rb2d;
    
    private Animator blackScreenAnimator;
    private SaveManager saveManager;
    private bool isInDangerPlatform = false;
    private bool isInHealPlatform = false;
    private bool facingRight;
    private bool isGrounded;
    private bool gameOver = false;
    private float moveInput;
    private int extraJumps;
```

Notice here that I have group by access modifier, then by type, and made a group of static.

Reply:
Everything that is public is for either debug purposes or for editing from Unity. But grouping is ok.